sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/compiler/integration/JavadocIntegration.kt

```diff
@@ -0,0 +1,38 @@
+package com.squareup.sqldelight.core.compiler.integration
+
+import com.intellij.psi.PsiElement
+
+/**
+ * This pattern consists of 3 parts:
+ *
+ * - `/\\*\\*` - matches the first line of the Javadoc:
+ *
+ * ```
+ * </**>
+ *  * Javadoc
+ *  */
+ * ```
+ *
+ * - `\n \\*[ /]?` - matches every other line of Javadoc:
+ *
+ * ```
+ * /**<
+ *  * >Javadoc<
+ *  */>
+ * ```
+ *
+ * - ` \\**slash` - specifically matches the tail part of a single-line Javadoc:
+ *
+ * ```
+ * /* Javadoc< */>
+ * ```
+ */
+private val JAVADOC_TEXT_REGEX = Regex("/\\*\\*|\n \\*[ /]?| \\*/")
+
+internal fun javadocText(javadoc: PsiElement?): String? {
+  if (javadoc == null) return null
```

Comment:
You already have a null check at the call site. The parameter can be non-null and get rid of this guard.